File: day1/1_2.py
def repeat_frequency(changes):
    s = 0
    seen_vals = set([0])
    while True:
        for n in changes:
            s += n
            if s in seen_vals:
                return s
            seen_vals.add(s)

repeat_frequency([int(i) for i in open('input1.txt', 'r')])

File: day1/1_2.go
package main

import (
    "io/ioutil"
    "strconv"
    "container/ring"
    "strings"
)

func dup(x string)(sum int) {
    list := strings.Split(x, "\n")
    r := ring.New(len(list))
    seen := map[int]bool{0: true}
    for i := 0; i < r.Len(); i++ {
        num, err := strconv.Atoi(list[i])
        if err != nil {
            panic(err)
        }
        r.Value = num
        r = r.Next()
    }
    for true {
        sum += r.Value.(int)
        if (seen[sum] == true) {
            return sum
        }
        seen[sum] = true
        r = r.Next()
    }
    return
}

func main() {
    f, err := ioutil.ReadFile("input1.txt")
    if err != nil {
        panic(err)
    }
    dup(string(f))
}